diff --git a/src/laserMapping.cpp b/src/laserMapping.cpp
--- a/src/laserMapping.cpp
+++ b/src/laserMapping.cpp
@@ -697,7 +697,7 @@ int main(int argc, char **argv)
             publish_frame_body(pubLaserCloudFull_body);
             publish_effect_world(pubLaserCloudEffect);
             publish_odometry(pubOdomAftMapped);
-            // publish_map(pubLaserCloudMap);
+            publish_map(pubLaserCloudMap);
             if (path_en)
                 publish_path(pubPath);
         }
