diff --git a/CMakeLists.txt b/CMakeLists.txt
index e151147..992c6cd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -84,9 +84,11 @@ catkin_package(
 )
 
 add_executable(sfastlio_mapping src/laserMapping.cpp include/ikd-Tree/ikd_Tree.cpp src/preprocess.cpp)
+add_dependencies(sfastlio_mapping ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
 target_link_libraries(sfastlio_mapping ${catkin_LIBRARIES} ${PCL_LIBRARIES} ${PYTHON_LIBRARIES} ${Sophus_LIBRARIES})
 target_include_directories(sfastlio_mapping PRIVATE ${PYTHON_INCLUDE_DIRS})
 
 add_executable(fastlio_mapping_re src/laserMapping_re.cpp include/ikd-Tree/ikd_Tree.cpp src/preprocess.cpp)
+add_dependencies(fastlio_mapping_re ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
 target_link_libraries(fastlio_mapping_re ${catkin_LIBRARIES} ${PCL_LIBRARIES} ${PYTHON_LIBRARIES} ${Sophus_LIBRARIES})
 target_include_directories(fastlio_mapping_re PRIVATE ${PYTHON_INCLUDE_DIRS})
